src/core/DynaRec_x64/instructions.cc

```diff
@@ -683,8 +683,67 @@ void DynaRecCPU::recMULTU() {
     gen.mov(dword[contextPointer + HI_OFFSET], edx);
 }
 
+template <int size, bool signExtend>
+void DynaRecCPU::recompileLoadWithDelay(LoadDelayDependencyType type) {
+    if (m_regs[_Rs_].isConst()) {
+        gen.mov(arg1, m_regs[_Rs_].val + _Imm_);
+    } else {
+        allocateReg(_Rs_);
+        gen.moveAndAdd(arg1, m_regs[_Rs_].allocatedReg, _Imm_);
+    }
+
+    switch (size) {
+        case 8:
+            call(psxMemRead8Wrapper);
+            break;
+        case 16:
+            call(psxMemRead16Wrapper);
+            break;
+        case 32:
+            call(psxMemRead32Wrapper);
+            break;
+        default:
+            PCSX::g_system->message("Invalid size for memory load in dynarec. Instruction %08x\n", m_psxRegs.code);
+            break;
+    }
+
+    if (_Rt_) {
+        PCSX::g_system->printf("Need delayed load @ %08X\nJIT buffer index: %08X\n", m_pc, gen.getSize());
```

Comment:
Won't this one spam a lot?

Reply:
No because it very rarely gets triggered, but it's also debug logging that will be removed in the final PR